aiosmtpd/controller.py

```diff
@@ -8,13 +8,16 @@
 
 @public
 class Controller:
-    def __init__(self, handler, loop=None, hostname='::0', port=8025):
+    def __init__(self, handler, loop=None, hostname='::0', port=8025,
+                 thread_timeout=None):
```

Comment:
Thanks for the response, you bring up some good points.  What do you think about this:

* Instead of hard coding 5 seconds in the constructor, what about falling back to an environment variable, say `$AIOSMTPD_CONTROLLER_TIMEOUT` which would be defined as a float number of seconds.
* If the envar isn't set, then use a short 1 second timeout.

This way the default behavior will be to not hang (hanging is bad!) but it won't wait too long before it times out.  OTOH, if we find in our CI or some downstream CI that the default is too short, then you could set the envar to increase it for your CI/tox or whatever run without having to modify your client code.

Then the question is whether it even makes sense to pass the timeout into the `Controller` constructor.